app/main.py

```diff
@@ -45,14 +45,10 @@ def __eq__(self, other) -> bool:
         return self.km == other
 
     def __le__(self, other) -> bool:
-        if isinstance(other, Distance):
-            return self.km <= other.km
-        return self.km <= other
+        return not self.__gt__(other)
```

Comment:
great, but we call magic methods with special interfaces (symbols for example)
`+` will call `__add__` method